File: frontend/src/lib/api/ai.ts
import { apiClient } from "@/lib/api-client";
import type { GenerateTaglinesResponse, GenerateImageResponse, GenerateStoryboardResponse } from "@/types/ai";

export async function generateTaglines(
  product: string,
  audience: string,
  tone: string,
  count = 5
): Promise<string[]> {
  const res = await apiClient.post<GenerateTaglinesResponse>("/api/ai/generate-taglines", {
    product,
    audience,
    tone,
    count,
  });
  return res.taglines;
}

export async function generateImage(
  promptContext: string,
  artifactType: string,
  tone: string,
  aspectRatio: string
): Promise<GenerateImageResponse> {
  return apiClient.post<GenerateImageResponse>("/api/ai/generate-image", {
    prompt_context: promptContext,
    artifact_type: artifactType,
    tone,
    aspect_ratio: aspectRatio,
  });
}

export async function generateStoryboard(
  topic: string,
  keyMessage: string,
  product: string,
  tone: string
): Promise<GenerateStoryboardResponse> {
  return apiClient.post<GenerateStoryboardResponse>("/api/ai/generate-storyboard", {
    topic,
    key_message: keyMessage,
    product,
    tone,
  });
}

File: backend/app/api/ai.py
from fastapi import APIRouter, Depends

from app.core.auth import get_current_user
from app.models.user import User
from app.schemas.ai import (
    GenerateTaglinesRequest,
    GenerateTaglinesResponse,
    GenerateImageRequest,
    GenerateImageResponse,
    GenerateStoryboardRequest,
    GenerateStoryboardResponse,
)
from app.services.ai_service import (
    generate_taglines,
    generate_image,
    generate_storyboard,
)

router = APIRouter(prefix="/api/ai", tags=["ai"])


@router.post("/generate-taglines", response_model=GenerateTaglinesResponse)
async def gen_taglines(
    data: GenerateTaglinesRequest,
    _current_user: User = Depends(get_current_user),
) -> GenerateTaglinesResponse:
    taglines = await generate_taglines(
        product=data.product,
        audience=data.audience,
        tone=data.tone,
        count=data.count,
    )
    return GenerateTaglinesResponse(taglines=taglines)


@router.post("/generate-image", response_model=GenerateImageResponse)
async def gen_image(
    data: GenerateImageRequest,
    _current_user: User = Depends(get_current_user),
) -> GenerateImageResponse:
    image_url, prompt_used = await generate_image(
        product=data.prompt_context,
        audience="",
        tone=data.tone,
        artifact_type=data.artifact_type,
        aspect_ratio=data.aspect_ratio,
    )
    return GenerateImageResponse(image_url=image_url, prompt_used=prompt_used)


@router.post("/generate-storyboard", response_model=GenerateStoryboardResponse)
async def gen_storyboard(
    data: GenerateStoryboardRequest,
    _current_user: User = Depends(get_current_user),
) -> GenerateStoryboardResponse:
    frames = await generate_storyboard(
        topic=data.topic,
        key_message=data.key_message,
        product=data.product,
        tone=data.tone,
    )
    return GenerateStoryboardResponse(frames=frames)
